src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -332,6 +337,42 @@ protected Signature(SignerInfo signerInfo, SignatureType type)
             return new SignatureVerificationSummary(Type, status, flags, timestamp, expirationTime, issues);
         }
 
+        public string GetSigningCertificateFingerprint(HashAlgorithmName algorithm)
+        {
+            if (!Enum.IsDefined(typeof(HashAlgorithmName), algorithm))
+            {
+                throw new ArgumentException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.UnrecognizedEnumValue,
+                        algorithm),
+                    nameof(algorithm));
+            }
+
+            var certificate = SignerInfo.Certificate;
+            if (certificate == null)
+            {
+                throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Strings.Verify_ErrorNoCertificate, FriendlyName));
+            }
+
+            if (_signingCertificateFingerprintLookup == null)
+            {
+                _signingCertificateFingerprintLookup = new Dictionary<HashAlgorithmName, string>();
+            }
+
+            if (_signingCertificateFingerprintLookup.TryGetValue(algorithm, out var fingerprint))
+            {
+                return fingerprint;
+            }
+
+            var certificateFingerprint = CertificateUtility.GetHash(certificate, algorithm);
+            fingerprint = BitConverter.ToString(certificateFingerprint).Replace("-", "");
```

Comment:
Seen this pattern a lot through our code, maybe there should be a utility for this?